js/ui/grid_core/ui.grid_core.virtual_scrolling.js

```diff
@@ -161,14 +164,19 @@ const VirtualScrollingDataSourceAdapterExtender = (function() {
             }
         },
         _handleLoadError: function() {
-            const that = this;
+            this._isLoading = false;
+            this.loadingChanged.fire(false);
 
-            that._isLoading = false;
-            that.loadingChanged.fire(false);
-
-            that.callBase.apply(that, arguments);
+            this.callBase.apply(this, arguments);
         },
         _handleDataChanged: function(e) {
+            if(this.option(NEW_SCROLLING_MODE)) {
+                this._items = this._dataSource.items();
```

Comment:
Try change items method to return _dataSource.items() directly instead of assign _items here.

Reply:
This already implemented. The code is redundant here.